### ISSUE TITLE
[STORM-1360] storm.py issues with python 3

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#5195]
<p>Python 3.x doesn't like the "print without ()" and "except with comma" syntax.</p>

---
<details><summary><i>Originally reported by <a title="Chuck Burgess" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ashnazg">ashnazg</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-1360" target="_blank">storm.py issues with python 3</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Chuck Burgess" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ashnazg">ashnazg</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2015-12-03T16:59:00+00:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>GitHub user ashnazg opened a pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911</a></p>

<p>    <a href="https://issues.apache.org/jira/browse/STORM-1360" title="storm.py issues with python 3" class="issue-link" data-issue-key="STORM-1360"><del>STORM-1360</del></a>:  storm.py issues with python 3</p>

<p>    This changes the "print" and "except" syntax usages such that they test fine for me on Python 2.6.6 (RHEL) and 3.5.0 (Win7).</p>

<p>You can merge this pull request into a Git repository by running:</p>

<p>    $ git pull <a href="https://github.com/ashnazg/storm" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/ashnazg/storm</a> python3</p>

<p>Alternatively you can review and apply these changes as the patch at:</p>

<p>    <a href="https://github.com/apache/storm/pull/911.patch" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911.patch</a></p>

<p>To close this pull request, make a commit to your master/trunk branch<br/>
with (at least) the following in the commit message:</p>

<p>    This closes #911</p>

<hr />
<p>commit 56b52a0e61d63bb5a9ea3e6d2c651d209bf889f8<br/>
Author: Chuck Burgess <cburgess@progressrail.com><br/>
Date:   2015-12-02T14:14:49Z</p>

<p>    make 'print' work with python2 and 3;</p>

<p>commit e92f3a6e7a67f9c67c5fdb3f588a377a47a94213<br/>
Author: Chuck Burgess <cburgess@progressrail.com><br/>
Date:   2015-12-02T14:15:11Z</p>

<p>    make 'except' work with python2 and 3;</p>

<hr />

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user unsleepy22 commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911#issuecomment-161324640" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911#issuecomment-161324640</a></p>

<p>    @ashnazg <br/>
    since python 3.x is officially not fully compatible with 2.x, is it necessary to make this change? By making this PR, we may need to check all python files.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user ashnazg commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911#issuecomment-161325344" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911#issuecomment-161325344</a></p>

<p>    I intentionally tested it against 2.x to be sure it was still ok, so that a larger scale effort would not necessarily be required.  I simply couldn't get my Win7 builds to run, due to the syntax breakdown as seen in the clj multilang-test.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user ashnazg commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911#issuecomment-161326961" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911#issuecomment-161326961</a></p>

<p>    As an aside, I had tried to find a reference in the Storm documentation with regard to Python version requirements, but did not see any.  That's why I explored why this file was failing for me on 3.x.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user ashnazg commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911#issuecomment-161329756" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911#issuecomment-161329756</a></p>

<p>    I'll have to rebase after #912 merges before this one will test good on Travis.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user revans2 commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911#issuecomment-161362063" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911#issuecomment-161362063</a></p>

<p>    @ashnazg the changes look good to me, and I would love to maintain both python2 and python3 compatibility.  I think the best way to do this is to have travis CI handle that for us.</p>

<p>    <a href="https://docs.travis-ci.com/user/languages/python" class="external-link" target="_blank" rel="nofollow noopener">https://docs.travis-ci.com/user/languages/python</a></p>

<p>    Would you mind filing a JIRA to have travis run with different versions of python. Ideally it would be nice to configure it so we only run the multi-lang tests more than once, but that could be an optimization that comes later on.</p>

<p>    Also it would be good to update to docs to indicate the versions of python supported.</p>

<p>    I am +1 for this change.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user ashnazg commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911#issuecomment-161386724" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911#issuecomment-161386724</a></p>

<p>    @revans2 , created <a href="https://issues.apache.org/jira/browse/STORM-1362" title="Travis should run at least the multilang-test using Python2 and 3" class="issue-link" data-issue-key="STORM-1362"><del>STORM-1362</del></a> for it.  I'll take a stab at it, but someone more familiar with Travis might can knock it out faster.  I don't see an obvious "travis-ish" way to do this, so I'm guessing it'll require its own shellscript.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user revans2 commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911#issuecomment-161411832" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911#issuecomment-161411832</a></p>

<p>    @ashnazg Ya I think we are going to need to do something special to make it work too, but travis is the place we should be doing it if we want to be sure we don't break things in the future.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user ashnazg commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/911#issuecomment-161715238" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911#issuecomment-161715238</a></p>

<p>    Not sure how I had this succeeding previously, but now it consistently does not.  What I'm seeing now implies that only a full-on python3 adjustment over many files solve this.  I'll just have to drop my Win7 env down to python2.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user ashnazg closed the pull request at:</p>

<p>    <a href="https://github.com/apache/storm/pull/911" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/911</a></p>
